### PR TITLE
Decouple infra-provider from core

### PR DESCRIPTION
**Pull request by cricket007**
# stream-registry PR

Split out `infra-provider` from `core` and refactor `infra-provider-kafka` to use it

### Added
* `infra-provider` module

### Changed
* Rename & refactor `kafka-infra-provider`

### Deleted
* Unused Jersey module in the schema events module

# PR Checklist Forms

- [x] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: infra-provider/src/main/java/com/homeaway/streamingplatform/provider/InfraManager.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    void start() throws Exception;
+
+    void stop() throws Exception;
+
+    /**
+     * Provides the {@link InfraManager} class with the given values for later configuration.
+     *
+     * @param configs the configurations specific for the {@link InfraManager}
+     */
+    void configure(Map<String, Object> configs);
```
[the remark this answers is not in the document]
Comment (cricket007) [reply infra-provider/src/main/java/com/homeaway/streamingplatform/provider/InfraManager.java]: Just copying the existing implementation...

I think we tried `String,String`, but the DropWizard YAML parser ended up wanting a `Map<String, ?>` and we just put Object there instead

[the remark this answers is not in the document]
Comment (neoword) [reply infra-provider/src/main/java/com/homeaway/streamingplatform/provider/InfraManager.java]: Ok.

File: infra-provider-kafka/src/main/java/com/homeaway/streamingplatform/provider/impl/KafkaInfraManager.java [hunk 2]
```diff
@@ -36,56 +37,44 @@
...
         Validate.validState(configs.containsKey(INFRA_KSTREAM_PROPS), "InfraKStreams properties is not provided.");
         Map<String, Object> infraKStreamsPropertiesMap = (Map<String, Object>) configs.get(INFRA_KSTREAM_PROPS);
         infraKStreamsPropertiesMap.forEach(infraKStreamsProperties::put);
-
         log.info("Infra KStreams Properties: {}", infraKStreamsProperties);
 
         // initialize the kstreams processor
-        infraKStreamBuilder.globalTable(infraManagerTopic, infraStateStoreName);
+        KStreamBuilder infraKStreamBuilder = new KStreamBuilder();
+        kTable = infraKStreamBuilder.globalTable(infraManagerTopic, infraStateStoreName);
```
Comment (neoword) [inline infra-provider-kafka/src/main/java/com/homeaway/streamingplatform/provider/impl/KafkaInfraManager.java]: Does this require proper closing during graceful shutdown?
Do any of these require proper closing during graceful shutdown?

If yes, add a comment and open an issue (lets not block this PR)

Comment (cricket007) [reply infra-provider-kafka/src/main/java/com/homeaway/streamingplatform/provider/impl/KafkaInfraManager.java]: Well, there is no close method on `GlobalKTable` interface (doesn't implement `Closeable` either). 

It seems `KafkaStreams.close()` shutsdown all revelant threads and objects.

**Pull request merged**